lib/puppet/provider/archive/default.rb

```diff
@@ -40,6 +41,20 @@ def archive_filepath
     resource[:path]
   end
 
+  def download(archive_filepath)
+    tempfile=Tempfile.new(resource[:name])
+    PuppetX::Bodeco::Util.download(resource[:source], tempfile.path, :username => resource[:username], :password => resource[:password], :cookie => resource[:cookie] )
+    tempfile.close
```

Comment:
Be advised that this may delete the file if `tempfile` is finalized before you move it elsewhere.  I would recommend replacing lines 45-47 with this:
```ruby
tempfile=Tempfile.new(resource[:name])
tempfile.close!
PuppetX::Bodeco::Util.download(resource[:source], tempfile.path, :username => resource[:username], :password => resource[:password], :cookie => resource[:cookie] )
```
`Tempfile#close!` forces deletion of the file to happen immediately.